src/scales.js

```diff
@@ -236,17 +238,17 @@ export function exposeScales(scaleDescriptors) {
   };
 }
 
-function exposeScale({scale, type, label, percent, diverging, align, paddingInner, paddingOuter, transform, round}) {
+function exposeScale({scale, type, label, range, percent, diverging, align, paddingInner, paddingOuter, transform, round}) {
   if (type === "identity") return {type: "identity"};
   const domain = scale.domain();
   const pivot = diverging && domain.splice(1,1)[0];
   return {
     type,
     domain,
-    range: scale.range(),
+    ...range !== undefined && {range},
```

Comment:
This will need a defensive copy.

```suggestion
    ...range !== undefined && {range: range.slice()},
```